resolver/resolve.go

```diff
@@ -160,7 +160,7 @@ func checkCache(ctx context.Context, q *Query) *RRCache {
 		log.Tracer(ctx).Trace("resolver: serving from cache, requesting new")
 
 		// resolve async
-		module.StartMediumPriorityMicroTask(&mtAsyncResolve, func(ctx context.Context) error {
+		module.StartLowPriorityMicroTask(&mtAsyncResolve, func(ctx context.Context) error {
```

Comment:
IMHO they still have higher priority because if we keep them back because the system is doing some other low-prio things (like writing logs), the whole network will become slow. The whole resolving is heavily network-IO dependent and thus the go-routine will block anyway and provide resources for other stuff to do. Maybe we nee a new "MicroTask" type for that because IO is actually something the goroutine schedule is awesome in. Not sure if we apply some artificial perfomance limits here....

Reply:
I see your point that IO should not use the microtask system. In this case, performance would not be impacted, as these queries run after an (outdated) answer was already provided to the requesting process. These queries just refresh the cache.
I would proceed by making it a worker. Originally, I switched it to low priority because I suspected it was keeping logs from being written.

Reply:
:+1: